bc_obps/compliance/models/compliance_product.py

```diff
@@ -0,0 +1,53 @@
+from django.db import models
+from registration.models.time_stamped_model import TimeStampedModel
+from reporting.models.report_product import ReportProduct
+from simple_history.models import HistoricalRecords
+from .compliance_summary import ComplianceSummary
+from .rls_configs.compliance_product import Rls as ComplianceProductRls
+
+
+class ComplianceProduct(TimeStampedModel):
+    """Model to store per-product compliance data"""
+
+    compliance_summary = models.ForeignKey(
+        ComplianceSummary,
+        on_delete=models.PROTECT,
+        related_name="products",
+        db_comment="The compliance summary this product data belongs to",
+    )
+    report_product = models.ForeignKey(
+        ReportProduct,
+        on_delete=models.PROTECT,
+        related_name="compliance_products",
+        db_comment="The report product this compliance data is for",
+    )
+    annual_production = models.DecimalField(max_digits=20, decimal_places=4, db_comment="Annual production amount")
+    apr_dec_production = models.DecimalField(
+        max_digits=20, decimal_places=4, db_comment="April-December production amount"
+    )
+    emission_intensity = models.DecimalField(
+        max_digits=20, decimal_places=4, db_comment="Product weighted average emission intensity"
+    )
+    allocated_industrial_process_emissions = models.DecimalField(
+        max_digits=20, decimal_places=4, db_comment="Allocated industrial process emissions in tCO2e"
+    )
+    allocated_compliance_emissions = models.DecimalField(
+        max_digits=20, decimal_places=4, db_comment="Allocated emissions for compliance in tCO2e"
+    )
+
+    history = HistoricalRecords(
+        table_name='erc_history"."compliance_product_history',
+        history_user_id_field=models.UUIDField(null=True, blank=True),
+    )
```

Comment:
I don't think this model needs historical tables, since it is created from a single, submitted, immutable report version.
We could also protect it behind an immutability trigger